qt/scientific_interfaces/Indirect/IndirectFitPlotModel.cpp

```diff
@@ -276,8 +277,21 @@ bool IndirectFitPlotModel::canCalculateGuess() const {
     return false;
 
   const auto composite = std::dynamic_pointer_cast<CompositeFunction>(function);
+  const auto resolutionLoaded = isResolutionLoaded();
   const auto isEmptyModel = composite && composite->nFunctions() == 0;
-  return getWorkspace() && !isEmptyModel;
+  return getWorkspace() && !isEmptyModel && resolutionLoaded;
+}
+
+bool IndirectFitPlotModel::isResolutionLoaded() const {
+  const auto model = dynamic_cast<ConvFitModel *>(m_fittingModel);
+  if (model) {
+    if (m_fittingModel->getResolutionsForFit().size() != 0) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+  return true;
```

Comment:
Is the resolution guarenteed to be loaded if there is no model?

Reply:
> 
> 
> Is the resolution guarenteed to be loaded if there is no model?


So if the cast fails it means its a different fit model (used by the other tabs), which don't require a resolution file. Granted this is a bit ambiguous, but it needs to return true for the sake of the comparisons its makes above. But currently can’t think of a better function name.

Reply:
That makes sense, thanks. Could you add a comment before L294 clarifying why it returns true? Without some context it looks a little counterintuitive.